servatrice/src/servatrice_database_interface.cpp

```diff
@@ -291,11 +291,48 @@ bool Servatrice_DatabaseInterface::checkUserIsBanned(const QString &ipAddress, c
     }
 
     return
-        checkUserIsIpBanned(ipAddress, banReason, banSecondsRemaining)
-        || checkUserIsNameBanned(userName, banReason, banSecondsRemaining);
+        checkUserIsIpBanned(ipAddress, banReason, banSecondsRemaining) || checkUserIsNameBanned(userName, banReason, banSecondsRemaining) || checkUserIsIdBanned(clientId, banReason, banSecondsRemaining);;
 
 }
 
+bool Servatrice_DatabaseInterface::checkUserIsIdBanned(const QString &clientId, QString &banReason, int &banSecondsRemaining)
+{
+    if (clientId.isEmpty())
+        return false;
+
+    QSqlQuery *idBanQuery = prepareQuery(
+            "select"
+                    " timestampdiff(second, now(), date_add(b.time_from, interval b.minutes minute)),"
+                    " b.minutes <=> 0,"
+                    " b.visible_reason"
+                    " from {prefix}_bans b"
+                    " where"
+                    " b.time_from = (select max(c.time_from)"
+                    " from {prefix}_bans c"
+                    " where c.clientid = :id)"
+                    " and b.clientid = :id2");
+
+    idBanQuery->bindValue(":id", clientId);
```

Comment:
I think what Zach means is, can we reuse `:id` so that it's `where c.clientid = :id and b.clientid = :id` and bind `:id` once?

I don't know the answer.